rudder/stream/message.go

```diff
@@ -0,0 +1,62 @@
+package stream
+
+import "encoding/json"
+
+const (
+	pulsarKeyMessageID       = "messageID"
+	pulsarKeyRoutingKey      = "routingKey"
+	pulsarKeyWorkspaceID     = "workspaceID"
+	pulsarKeySourceID        = "sourceID"
+	pulsarKeyUserID          = "userID"
+	pulsarKeySourceJobRunID  = "sourceJobRunID"
+	pulsarKeySourceTaskRunID = "sourceTaskRunID"
+	pulsarKeyTraceID         = "traceID"
+)
+
+type Message struct {
+	Properties MessageProperties `json:"properties"`
+	Payload    json.RawMessage   `json:"payload"`
+}
+
+type MessageProperties struct {
+	MessageID       string `json:"messageID"`
+	RoutingKey      string `json:"routingKey"`
+	WorkspaceID     string `json:"workspaceID"`
+	UserID          string `json:"userID"`
+	SourceID        string `json:"sourceID"`
+	SourceJobRunID  string `json:"sourceJobRunID"`  //optional
+	SourceTaskRunID string `json:"sourceTaskRunID"` //optional
+	TraceID         string `json:"traceID"`
+}
+
+// FromPulsarMessage converts a Pulsar message to a Message.
+func FromPulsarMessage(properties map[string]string, payload []byte) (Message, error) {
```

Comment:
Should we complain here if a mandatory property is missing? We could consider having some basic validations here to avoid repetition of the same thing in each service. I added a similar comment below, let me know wyt.

Reply:
Yes, we can do that as well :+1: